airflow/jobs/local_task_job.py

```diff
@@ -226,6 +226,18 @@ def heartbeat_callback(self, session=None):
     @Sentry.enrich_errors
     def _run_mini_scheduler_on_child_tasks(self, session=None) -> None:
         try:
+
+            if (
+                self.task_instance.task.wait_for_downstream
+                and self.task_instance.get_previous_ti()
+                and not self.task_instance.are_dependents_done()
+            ):
+                self.log.info(
+                    "No downstream tasks scheduled because task instance "
+                    "dependents are still running and wait_for_downstream is true"
```

Comment:
```suggestion
                    "dependents have not completed yet and wait_for_downstream is true"
```

as the task can be in queued or scheduled state too